evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -0,0 +1,1146 @@
+
+    component_graphs = {}
+    for ind, estimator in enumerate(estimators):
+        component_graphs[f"CG_{ind}"] = [estimator]
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="multiclass",
+        allowed_component_graphs=component_graphs,
+        custom_hyperparameters=custom_hyperparameters,
+        n_jobs=1,
+        _automl_algorithm="iterative",
+    )
+    env = AutoMLTestEnv("multiclass")
+
+    mock_add.side_effect = ValueError("Alternate error that can be thrown")
+    with pytest.raises(ValueError) as error:
+        with env.test_context(score_return_value={"Log Loss Multiclass": 1.0}):
+            automl.search()
+    assert "Alternate error that can be thrown" in str(error.value)
+    assert "Default parameters for components" not in str(error.value)
+
+
+@pytest.mark.parametrize("component_graphs", [True, False])
+@pytest.mark.parametrize("automl_parameters", [True, False])
+@pytest.mark.parametrize("custom_hyperparameters", [True, False])
+def test_pipeline_custom_hyperparameters_make_pipeline_iterative(
+    custom_hyperparameters,
+    automl_parameters,
+    component_graphs,
+    X_y_multi,
+    AutoMLTestEnv,
+):
+    X, y = X_y_multi
+    X = pd.DataFrame(X, columns=[f"Column_{i}" for i in range(20)])
+
+    component_graph_ = None
+    automl_parameters_ = None
+    custom_hyperparameters_ = None
+
+    if component_graphs:
+        component_graph_ = {
+            "Name_0": [
+                "Drop Columns Transformer",
+                "Imputer",
+                "Random Forest Classifier",
+            ]
+        }
+
+    if automl_parameters:
+        automl_parameters_ = {
+            "Drop Columns Transformer": {
+                "columns": ["Column_0", "Column_1", "Column_2"]
+            },
+            "Random Forest Classifier": {"n_estimators": 201},
+        }
+    if custom_hyperparameters:
+        custom_hyperparameters_ = {
+            "Imputer": {"numeric_impute_strategy": Categorical(["mean"])},
+            "Random Forest Classifier": {
+                "max_depth": Integer(4, 7),
+                "n_estimators": Integer(190, 210),
+            },
+        }
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="multiclass",
+        allowed_component_graphs=component_graph_,
+        pipeline_parameters=automl_parameters_,
+        custom_hyperparameters=custom_hyperparameters_,
+        max_batches=4,
+        _automl_algorithm="iterative",
+    )
+    env = AutoMLTestEnv("multiclass")
+    with env.test_context(score_return_value={"Log Loss Multiclass": 1.0}):
+        automl.search()
+
+    for i, row in automl.full_rankings.iterrows():
+        if "Random Forest Classifier" in row["pipeline_name"]:
+            if component_graph_ and automl_parameters:
+                assert row["parameters"]["Drop Columns Transformer"]["columns"] == [
```

Comment:
Same coverage issues here.

Reply:
This was originally called `test_iterative_algorithm_pipeline_custom_hyperparameters_make_pipeline` and this coverage is missing on main as well! https://app.codecov.io/gh/alteryx/evalml/blob/main/evalml/tests/automl_tests/test_automl.py